recipes/mosquitto/2.x/conanfile.py

```diff
@@ -40,32 +42,25 @@ class Mosquitto(ConanFile):
         "websockets": False,
         "threading": True,
     }
-    exports_sources = ["CMakeLists.txt"]
-    generators = "cmake", "cmake_find_package"
-    _cmake = None
-
-    @property
-    def _source_subfolder(self):
-        return "source_subfolder"
-
-    @property
-    def _build_subfolder(self):
-        return "build_subfolder"
+    generators = "CMakeDeps"
 
     def config_options(self):
         if self.settings.os == "Windows":
-            del self.options.fPIC
+            self.options.rm_safe("fPIC")
 
     def configure(self):
         if self.options.shared:
-            del self.options.fPIC
+            self.options.rm_safe("fPIC")
         if not self.options.clients:
-            del self.options.cjson
+            self.options.rm_safe("cjson")
         if not self.options.broker:
-            del self.options.websockets
+            self.options.rm_safe("websockets")
         if not self.options.build_cpp:
-            del self.settings.compiler.libcxx
-            del self.settings.compiler.cppstd
+            self.options.rm_safe("compiler.libcxx")
+            self.options.rm_safe("compiler.cppstd")
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
 
     def requirements(self):
         if self.options.ssl:
```

Comment:
You can change the openssl version to the range version `[>=1.1 <4]` 

For Cmake and OpenSSL we are adopting ranges: https://github.com/conan-io/conan-center-index/blob/master/docs/adding_packages/dependencies.md#version-ranges